common/unsafe/src/main/java/org/apache/spark/sql/catalyst/util/CollationAwareUTF8String.java

```diff
@@ -34,6 +34,15 @@
  * Utility class for collation-aware UTF8String operations.
  */
 public class CollationAwareUTF8String {
+
+  public static boolean lowercaseMatchAt(final UTF8String l, final UTF8String r,
+      int pos, int len) {
+    if (len + pos > l.numChars() || pos < 0) {
+      return false;
+    }
```

Comment:
Would we ever pass a negative `pos`?

Reply:
after all, this method is part of a public interface (CollationAwareUTF8String.java) - so I think this check is in order (we don't know who or how may use this method exactly, although we currently only pass positive `pos` values to it)